spec/controllers/allies_controller_spec.rb

```diff
@@ -74,10 +79,7 @@
         end
 
         it 'deletes the allyship request notification' do
-          allow(Notification).to receive(:where).and_return(notification)
-          allow(notification).to receive(:order).and_return(notification)
-          expect(notification).to receive(:destroy_all)
-          subject
+          expect { subject }.to change { user.notifications.count }.from(1).to(0)
```

Comment:
👏 Nice! I like this refactor. This test is shorter more agnostic to how the method is implemented.